src/options.rs

```diff
@@ -1,12 +1,24 @@
+use crate::collector::Input;
+
 use anyhow::{Error, Result};
+use lazy_static::lazy_static;
 use serde::Deserialize;
 use std::{fs, io::ErrorKind};
 use structopt::{clap::crate_version, StructOpt};
 
 pub(crate) const USER_AGENT: &str = concat!("lychee/", crate_version!());
 const METHOD: &str = "get";
-const TIMEOUT: &str = "20";
-const MAX_CONCURRENCY: &str = "128";
+const TIMEOUT: usize = 20;
+const MAX_CONCURRENCY: usize = 128;
+const MAX_REDIRECTS: usize = 10;
+
+// this exists because structopt requires `&str` type values for defaults
+// (we can't use e.g. `TIMEOUT` or `timeout()` which gets created for serde)
+lazy_static! {
+    static ref TIMEOUT_STR: String = TIMEOUT.to_string();
+    static ref MAX_CONCURRENCY_STR: String = MAX_CONCURRENCY.to_string();
+    static ref MAX_REDIRECTS_STR: String = MAX_REDIRECTS.to_string();
+}
```

Comment:
I got dismayed by these values being `String`.

I think at some point we should overhaul the opts+config definition and parsing.

Reply:
oh yeah, it's getting a bit out of hand